### PR TITLE
DAOS-2370 pool: Use pool ownership props

### PR DESCRIPTION
**Pull request by kjacque**
This change is an incremental step toward removing owner
UID and GID from the pool creation API. The owner and
owner-group pool properties should be used instead.

- Remove owner uid/gid/mode from pools under the covers.
- Convert from owner strings to uid/gid in DFS/dfuse.
- Convert uid/gid passed by client to daos_pool_create into
  ownership properties on pool creation, if not supplied as
  optional props.
- Use the ownership properties for checking access during
  pool connect.

Signed-off-by: Kris Jacque <kristin.jacque@intel.com>

### CODE REVIEW

File: src/client/api/mgmt.c [hunk 1]
```diff
@@ -112,24 +124,82 @@ daos_pool_create(uint32_t mode, uid_t uid, gid_t gid, const char *grp,
 		return -DER_INVAL;
 	}
 
+	entries = (pool_prop == NULL) ? 0 : pool_prop->dpp_nr;
+
+	/*
+	 * TODO: remove uid/gid input params and use euid/egid instead
```
Comment (dpquigl) [inline src/client/api/mgmt.c]: I think its ok to use this like this, You may want to create a pool with a uid/gid not yours. In the current dmg tool they make the decision whether to pull euid/egid or to use what is passed in on the command line.

File: src/client/dfuse/dfuse_pool.c [hunk 2]
```diff
@@ -103,20 +106,45 @@ dfuse_pool_lookup(fuse_req_t req, struct dfuse_inode_entry *parent,
...
+	rc = daos_acl_principal_to_gid(prop_entry->dpe_str,
+				       &ie->ie_stat.st_gid);
+	if (rc != 0) {
+		DFUSE_LOG_ERROR("Unable to convert owner-group to gid: (%d)",
+				rc);
+		D_GOTO(close, rc);
+	}
 
-	/* TODO: This should inspect pi_mode and corrrectly construct a correct
+	/* TODO: This should inspect ACLs and correctly construct a correct
```
Comment (dpquigl) [inline src/client/dfuse/dfuse_pool.c]: This should really only inspect owner_user owner_group and everyone.

Comment (kjacque) [reply src/client/dfuse/dfuse_pool.c]: Yeah, that approach makes sense to me. Going to add folks from the dfuse team once CI stabilizes, so they are aware of this.

File: src/client/api/mgmt.c (hunk 1 above)
Comment (kjacque) [reply src/client/api/mgmt.c]: In that case the owner user/group should be passed in as part of the pool_prop, as strings. It's only if nothing is passed in to us that we should map it to the euid/egid. I believe this entry point actually goes away altogether once the new DMG tool is done, so the TODO applies only to the short-term usage of this API.

**Comment by kjacque** [comment]
This PR depends on #475. The base branch will need to be changed to master after #475 lands.

File: src/client/api/mgmt.c [hunk 3]
```diff
@@ -112,24 +124,82 @@ daos_pool_create(uint32_t mode, uid_t uid, gid_t gid, const char *grp,
...
-	args->prop	= pool_prop;
+	args->prop	= final_prop;
 	args->svc	= svc;
 	args->uuid	= uuid;
 
-	return dc_task_schedule(task, true);
+	rc = dc_task_schedule(task, true);
+
+out:
+	daos_prop_free(final_prop);
```
Comment (mchaarawi) [inline src/client/api/mgmt.c]: If the call to daos_pool_create is done asynchronously, that free call would make args->prop invalid and will cause segfaults in the dc_pool_create, no? I think you will need to free it in the completion callback of the pool create operation (pool_create_cp in src/mgmt/cli_pool.c)
I'm not sure about owner or owner_grp if they are used internally or not, in which case they need to be freed after the operation is done too.

File: src/include/daos_task.h [hunk 4]
```diff
@@ -142,9 +142,6 @@ typedef struct {
 } daos_set_params_t;
 
 typedef struct {
-	uint32_t		mode;
```
Comment (mchaarawi) [inline src/include/daos_task.h]: Those structs should always mirror what the actual API looks like, so you can't remove mode/uid/gid if those remain in the API. Those are ways where we can call the DAOS API in a different manner than the daos_xyz() functions.

Comment (kjacque) [reply src/include/daos_task.h]: Fixed

File: src/client/api/mgmt.c (hunk 3 above)
Comment (kjacque) [reply src/client/api/mgmt.c]: Fixed

**Pull request merged**